service/src/main/java/uk/gov/hmcts/reform/fpl/service/CaseExtensionService.java

```diff
@@ -0,0 +1,37 @@
+package uk.gov.hmcts.reform.fpl.service;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+import uk.gov.hmcts.reform.fpl.model.CaseData;
+
+import java.time.LocalDate;
+import java.util.Optional;
+
+@Service
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class CaseExtensionService {
+    private final ObjectMapper mapper;
+
+    public LocalDate getCaseCompletionDate(CaseDetails caseDetails) {
```

Comment:
Can we pass CaseData as an argument here, so we don't have to use ObjectMapper? That'd require adding used case fields to CaseData.